EmailScraper.java

```diff
@@ -0,0 +1,163 @@
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**                                                                                                          
+ * Takes an internet domain name and prints out a list of email addresses on                                 
+ * that website only. The program should not crawl other subdomains (for                                     
+ * example, given "jana.com", it should not crawl blog.jana.com or                                           
+ * technology.jana.com).                                                                                     
+ *                                                                                                           
+ * Note that this means that specifying "jana.com" will NOT crawl links                                      
+ * specified as www.jana.com; www is the conventional subdomain, but it                                      
+ * is a different subdomain.                                                                                 
+ *                                                                                                           
+ * These are expected output from www.jana.com                                                               
+ * Found these email addresses:                                                                              
+ * sales@jana.com                                                                                            
+ * press@jana.com                                                                                            
+ * info@jana.com                                                                                             
+ *                                                                                                           
+ * https://gist.github.com/clee-jana/a1535a7ea0c7b0fed41d                                                    
+ *                                                                                                           
+ * The email addresses should be all human visible text that looks like an email                             
+ * address, not in comments, inline JavaScript, and similar invisible text.                                  
```

Comment:
I'd need to run a JavaScript interpreter, which is possible (the Google crawler certainly does it) but non trivial. Here is a thread discussing the issues and offering a few possibilities. It would take more than a 2-4 hour coding challenge project, at least for the first time.